### PR TITLE
Add subproject for shading dependencies

### PR DESCRIPTION
**Pull request by calvin681**
### Context

Explain context and other details for this pull request.

### Checklist

- [ ] `./gradlew build` compiles code correctly
- [ ] Added new tests where applicable
- [ ] `./gradlew test` passes all tests
- [ ] Extended README or added javadocs where applicable
- [ ] Added copyright headers for new files from `CONTRIBUTING.md`

### CODE REVIEW

File: mantis-server/mantis-server-worker/build.gradle [hunk 1]
```diff
@@ -19,7 +19,7 @@ apply plugin: 'eu.appsatori.fatjar'
 ext {
     mantisControlPlaneVersion = '1.2.+'
     mantisRxControlVersion = '1.3.14'
-    mesosVersion = '1.7.2'
+    mesosVersion = '1.3.2'
```
Comment (jeffchao) [inline mantis-server/mantis-server-worker/build.gradle:22]: Looks like a rebase/merge error. This should be `1.7.2`.

File: mantis-common/src/main/java/com/mantisrx/common/utils/JsonUtility.java [hunk 2]
```diff
@@ -16,15 +16,14 @@
 
 package com.mantisrx.common.utils;
 
+import mantis.com.fasterxml.jackson.core.type.TypeReference;
```
Comment (jeffchao) [inline mantis-common/src/main/java/com/mantisrx/common/utils/JsonUtility.java:19]: We probably don't want to commandeer the `mantis` namespace globally. We could relocate packages into `io.mantisrx.shaded`.

File: mantis-shaded/build.gradle [hunk 3]
```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2019 Netflix, Inc.
```
Comment (jeffchao) [inline mantis-shaded/build.gradle:2]: 2020. You can add in copyright from the contributing directions https://github.com/Netflix/mantis/blob/master/CONTRIBUTING.md.

File: mantis-common/src/main/java/com/mantisrx/common/utils/JsonUtility.java (hunk 2 above)
Comment (calvin681) [reply mantis-common/src/main/java/com/mantisrx/common/utils/JsonUtility.java:19]: Makes sense. I was following the pattern from mantis-rxnetty. So we may need to relocate that if we decide to merge it into this shaded project in the future.

Comment (codyrioux) [reply mantis-common/src/main/java/com/mantisrx/common/utils/JsonUtility.java]: Ah, yeah I agree `io.mantisrx.shaded` is the place to put it.

File: mantis-common/src/main/java/io/mantisrx/server/core/ServiceRegistry.java [hunk 4]
```diff
@@ -43,7 +43,7 @@ public MantisPropertiesService getPropertiesService() {
     private MantisPropertiesLoader loadMantisPropertiesService() {
         MantisPropertiesLoader mpl = new DefaultMantisPropertiesLoader(new Properties());
         try {
-            mpl = (MantisPropertiesLoader) Class.forName("com.netflix.mantis.common.properties.MantisFastPropertiesLoader").getConstructor(Properties.class)
+            mpl = (MantisPropertiesLoader) Class.forName("com.netflix.io.mantisrx.shaded.common.properties.MantisFastPropertiesLoader").getConstructor(Properties.class)
```
Comment (jeffchao) [inline mantis-common/src/main/java/io/mantisrx/server/core/ServiceRegistry.java:46]: This is a new file. Not clear what's the right path here: `com.netflix.mantis` vs `com.netflix.io.mantisrx` vs `io.mantisrx`. Can you explain?

Comment (jeffchao) [reply mantis-common/src/main/java/io/mantisrx/server/core/ServiceRegistry.java:46]: cc @codyrioux

Comment (codyrioux) [reply mantis-common/src/main/java/io/mantisrx/server/core/ServiceRegistry.java:46]: I've never seen this but I think in general for the OSS stuff we'll want to stick to `io.mantisrx` as the root namespace.

Comment (calvin681) [reply mantis-common/src/main/java/io/mantisrx/server/core/ServiceRegistry.java:46]: Reverting this file. This creeped in from find/replace. This class only exists in our internal repo and is loaded dynamically. We will not shade that class.

**Pull request merged**